product_uom_updatable/models/product_template.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Acsone SA/NV
+# Copyright 2020 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from itertools import groupby
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = "product.template"
+
+    def write(self, vals):
+        uom_id = vals.pop("uom_id", False)
+        uom_po_id = vals.pop("uom_po_id", False)
+        if uom_id:
+            self._update_uom(uom_id, "uom_id")
+        if uom_po_id:
+            self._update_uom(uom_po_id, "uom_po_id")
+        res = super().write(vals)
+        return res
+
+    def _update_uom(self, uom_id, field_name):
+        uom_obj = self.env["uom.uom"]
+        sorted_items = sorted(self, key=lambda r: r[field_name])
+        for key, products_group in groupby(sorted_items, key=lambda r: r[field_name]):
+            product_ids = [product.id for product in products_group]
+            new_uom = uom_obj.browse(uom_id)
+            if (
+                key.category_id == new_uom.category_id
+                and key.factor_inv == new_uom.factor_inv
+            ):
+                # pylint: disable=sql-injection
+                query = (
+                    "UPDATE product_template SET "
+                    + field_name.strip("'")
+                    + " = %s WHERE id in \
+                     %s"
+                )
```

Comment:
I thought I had pointed you to https://www.psycopg.org/docs/sql.html#module-psycopg2.sql which is the psycopg2's way to compose queries and deal with sql injections safely.

That would be something like:

```suggestion
                query = sql.SQL(
                    "UPDATE product_template SET {field} "
                    " = %s WHERE id in %s"
                ).format(field=sql.Identifier(field_name))
```

BTW, you don't need `+` to append several strings together, this is automatically concatenated:

```
>>> abc = (
...     "a"
...     "b"
...     "c"
... )
>>>
>>> abc
'abc'
>>>
```